File: 프로그래머스/2/42578. 의상/의상.py
def solution(cloth):
    answer = 1
    clothes = dict()
    
    for n,c in cloth:
        if c not in clothes:
            clothes[c] = [n]
        else:
            clothes[c].append(n)
    
    total_cnt = len(clothes)
    for c in clothes:
        tmp = len(clothes[c])
        if total_cnt == 1:
            return tmp
        answer *= tmp + 1
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, ArrayList<String>> map = new HashMap<>();
        
        for (String[] c: clothes) {
            String name = c[0];
            String category = c[1];
            
            if (map.containsKey(category))
                map.get(category).add(name);
            else {
                ArrayList<String> tmp = new ArrayList<>();
                tmp.add(name);
                map.put(category, tmp);
            }
        }
        
        for (String c: map.keySet()) {
            int count = map.get(c).size();
            answer *= count + 1;
        }
        return answer-1;
    }
}